php/OAuthSimple.php

```diff
@@ -490,9 +490,16 @@ private function _generateSignature ($parameters="")
 		{
             $secretKey .= self::_oauthEscape($this->_secrets['oauth_secret']);
         }
-        if(empty($parameters)){
-            $parameters = $this->_normalizedParameters();
+        
+        if(empty($parameters))
+        {
+            $parameters = urlencode($this->_normalizedParameters());
+        } 
+        else
+        {
+            $parameters = urlencode($parameters);
```

Comment:
Yep. I really need to go over this code again and do A LOT of clean-up. I prefer the second option, since that's less work for the CPU. (Thanks!)

Hopefully, I'll get some time this weekend and work on this.